src/abstractdataframe/abstractdataframe.jl

```diff
@@ -867,9 +867,19 @@ end
 
 """
     filter(function, df::AbstractDataFrame)
+    filter(cols => function, df::AbstractDataFrame)
 
 Return a copy of data frame `df` containing only rows for which `function`
-returns `true`. The function is passed a `DataFrameRow` as its only argument.
+returns `true`.
+
+If `cols` is not specified then the function is passed `DataFrameRow`s.
+If `cols` is a single column then then elements of column `cols` are passed to the function.
+If `cols` represents multiple columns (any valid column selector is accepted) then
```

Comment:
This should be kept consistent with `by`/`combine`/`select` (https://github.com/JuliaData/DataFrames.jl/issues/1935): if we decide to splat, we should also do it here. But I guess we can merge the PR as-is, and change the code before the next breaking release if we decide to change other functions too.

Reply:
Agreed. We can wait a few days with merging this I think (when we get the response on #data).

Reply:
I will rewrite it. We will auto-splat